### DISCUSSION TITLE
tutorials/navigation/navigation_introduction_2d

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#222] [User-contributed notes]
# tutorials/navigation/navigation_introduction_2d

Godot provides multiple objects, classes and servers to facilitate grid-based or mesh-based navigation and pathfinding for 2D and 3D games. The following section provides a quick overview over all ...

https://docs.godotengine.org/en/stable/tutorials/navigation/navigation_introduction_2d.html

**Comment by R41Ryan** [comment]
I've found that, referring to NavigationAgent2D, you can't set **target_position** and then call **get_next_path_position()** in the same _physics_process() call.

If you do want to set the **target_position** and call **get_next_path_position()** in the same frame (like say if the target_position is constantly changing), set **target_position** in a _process() call, and call **get_next_path_position()** in a _physics_process() call

**Comment by boholstapp** [comment]
Video Example
https://youtu.be/qIqcp7xBGkw
